File: frontend/novalite-todo-portal/src/app/shared/models/todo-list.ts
import { todoItem } from "./todo-item";

export interface todoList {
    id: string;
    title: string;
    description: string;
    todoItems: todoItem[]
}

export interface updateTodoList{
    id: string;
    title: string;
    description: string;
}

export interface newTodoList{
    title:string;
    description:string;
    todoUserId: string;
}

File: backend/NovaLite.Todo.Apii/NovaLite.Todo.Core/Models/TodoList.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NovaLite.Todo.Core.Models
{
    public class TodoList
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        [Required]
        [StringLength(255)]
        public string Title { get; set; } = string.Empty;
        [Required]
        public string Description { get; set; } = string.Empty;
        public Guid TodoUserId { get; set; }
        public TodoUser TodoUser { get; set; } = null!;
        public ICollection<TodoItem> TodoItems { get; set; } = new List<TodoItem>();
        public ICollection<TodoReminder> TodoReminder { get; set; } = new List<TodoReminder>();
        public ICollection<TodoAttachment> TodoAttachments { get; set; } = new List<TodoAttachment>();

    }
}
